src/main/java/org/elasticsearch/index/mapper/internal/IdFieldMapper.java

```diff
@@ -144,6 +146,8 @@ protected IdFieldMapper(String name, String indexName, float boost, FieldType fi
         super(new Names(name, indexName, indexName, name), boost, fieldType, docValues, Lucene.KEYWORD_ANALYZER,
                 Lucene.KEYWORD_ANALYZER, postingsProvider, docValuesProvider, null, null, fieldDataSettings, indexSettings);
         this.path = path;
+        this.writePre20Settings = indexSettings.equals(ImmutableSettings.EMPTY) == false &&
```

Comment:
Right I see... I think the `Version.indexCreated` should just return null in that case, but that is outside of the scope of this change.